### PR TITLE
Test size

### PR DESCRIPTION
**Pull request by icristescu**
Add an exception which is raised when trying to add keys and values of unexpected sizes.

### CODE REVIEW

File: src/index.ml [hunk 1]
```diff
@@ -80,8 +82,14 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
   let entry_sizeL = Int64.of_int entry_size
 
   let append_entry io e =
-    IO.append io (K.encode e.key);
-    IO.append io (V.encode e.value)
+    let encoded_key = K.encode e.key in
+    let encoded_value = V.encode e.value in
+    if String.length encoded_key != K.encoded_size then
```
Comment (samoht) [inline src/index.ml]: should be `<>` instead of `!=`

File: src/index.ml [hunk 2]
```diff
@@ -62,6 +62,8 @@ let may f = function None -> () | Some bf -> f bf
 
 exception RO_Not_Allowed
 
+exception Different_Sizes_Not_Allowed
```
Comment (pascutto) [inline src/index.ml]: What would you think of having exceptions like
```
exception Invalid_Key_Size of key
exception Invalid_Value_Size of value
```
to get better feedback ?

Comment (CraigFe) [reply src/index.ml]: Seems good to me. Would also make the `check_raises` tests slightly more precise.

**Comment by pascutto** [comment]
Thank you !

**Pull request merged**